src/Kiota.Builder/Refiners/ShellRefiner.cs

```diff
@@ -0,0 +1,180 @@
+            ConvertUnionTypesToWrapper(generatedCode, _configuration.UsesBackingStore);
+            AddRawUrlConstructorOverload(generatedCode);
+            AddPropertiesAndMethodTypesImports(generatedCode, false, false, false);
+            CreateCommandBuilders(generatedCode);
+            AddAsyncSuffix(generatedCode);
+            AddInnerClasses(generatedCode, false);
+            AddParsableInheritanceForModelClasses(generatedCode, "IParsable");
+            CapitalizeNamespacesFirstLetters(generatedCode);
+            ReplaceBinaryByNativeType(generatedCode, "Stream", "System.IO");
+            MakeEnumPropertiesNullable(generatedCode);
+            /* Exclude the following as their names will be capitalized making the change unnecessary in this case sensitive language
+             * code classes, class declarations, property names, using declarations, namespace names
+             * Exclude CodeMethod as the return type will also be capitalized (excluding the CodeType is not enough since this is evaluated at the code method level)
+            */
+            ReplaceReservedNames(
+                generatedCode,
+                new CSharpReservedNamesProvider(), x => $"@{x.ToFirstCharacterUpperCase()}",
+                new HashSet<Type> { typeof(CodeClass), typeof(CodeClass.Declaration), typeof(CodeProperty), typeof(CodeUsing), typeof(CodeNamespace), typeof(CodeMethod) }
+            );
+            DisambiguatePropertiesWithClassNames(generatedCode);
+            AddConstructorsForDefaultValues(generatedCode, false);
+            AddSerializationModulesImport(generatedCode);
+        }
+
+        private static void CreateCommandBuilders(CodeElement currentElement)
+        {
+            if (currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.RequestBuilder))
+            {
+                // Replace Nav Properties with BuildXXXCommand methods
+                var navProperties = currentClass.GetChildElements().OfType<CodeProperty>().Where(e => e.IsOfKind(CodePropertyKind.RequestBuilder));
+                foreach (var navProp in navProperties)
+                {
+                    var method = CreateBuildCommandMethod(navProp, currentClass);
+                    currentClass.AddMethod(method);
+                    currentClass.RemoveChildElement(navProp);
+                }
+
+                // Build command for indexers
+                var indexers = currentClass.GetChildElements().OfType<CodeIndexer>();
+                var classHasIndexers = indexers.Any();
+                foreach (var indexer in indexers)
+                {
+                    var method = new CodeMethod
+                    {
+                        Name = "BuildCommand",
+                        IsAsync = false,
+                        MethodKind = CodeMethodKind.CommandBuilder,
+                        OriginalIndexer = indexer
+                    };
+
+                    method.ReturnType = CreateCommandType(method);
```

Comment:
the return type setter already sets the parent now in case you want to simplify your code